service/src/main/java/uk/nhs/adaptors/gp2gp/TransformJsonToXml.java

```diff
@@ -0,0 +1,225 @@
+
+    private static final FhirParseService FHIR_PARSE_SERVICE = new FhirParseService();
+
+    public static void main(String[] args) throws Exception {
+
+        var inputWrapper = getFiles();
+
+        for (int i = 0; i < inputWrapper.getJsonFileInputs().size(); i++) {
+            var jsonString = inputWrapper.getJsonFileInputs().get(i);
+            String xmlResult = mapJsonToXml(jsonString);
+            var fileName = inputWrapper.getJsonFileNames().get(i);
+            writeToFile(xmlResult, fileName);
+        }
+    }
+
+    private static InputWrapper getFiles() {
+
+        File[] files = new File(JSON_FILE_INPUT_PATH).listFiles();
+        List<String> jsonStringInputs = new ArrayList<>();
+        List<String> fileNames = new ArrayList<>();
+
+        for (File inputJsonFile : files) {
+            LOGGER.info("Parsing File: " + inputJsonFile.getName());
+            if (FilenameUtils.getExtension(inputJsonFile.getName()).equalsIgnoreCase("json")) {
+                try {
+                    String jsonAsString = readJsonFileAsString(JSON_FILE_INPUT_PATH + inputJsonFile.getName());
+                    jsonStringInputs.add(jsonAsString);
+                    fileNames.add(inputJsonFile.getName());
+                } catch (Exception e) {
+                    LOGGER.info("Could Not Read Json Files.");
+                }
+            } else {
+                LOGGER.info("No .json Files Found");
+            }
+        }
+        return InputWrapper.builder().jsonFileInputs(jsonStringInputs).jsonFileNames(fileNames).build();
+    }
+
+    private static String extractNhsNumber(String json) {
+        var nhsnumberSystem = "https://fhir.nhs.uk/Id/nhs-number";
+        var bundle = FHIR_PARSE_SERVICE.parseResource(json, Bundle.class);
+        var nhsNumber = bundle.getEntry().stream()
+                .map(Bundle.BundleEntryComponent::getResource)
+                .filter(resource -> ResourceType.Patient.equals(resource.getResourceType()))
+                .map(resource -> (Patient) resource)
+                .map(resource -> (Identifier) getNhsNumberIdentifier(nhsnumberSystem, resource))
+                .findFirst().get().getValue();
+        return nhsNumber;
+    }
+
+    private static Object getNhsNumberIdentifier(String nhsnumberSystem, Patient resource) {
+        return resource.getIdentifier()
+                .stream().filter(identifier -> identifier.getSystem().equals(nhsnumberSystem)).findFirst().get();
+    }
+
+    private static String readJsonFileAsString(String file) throws Exception {
+        return new String(Files.readAllBytes(Paths.get(file)));
+    }
+
+    private static void writeToFile(String xml, String sourceFileName) {
+        String outputFileName = FilenameUtils.removeExtension(sourceFileName);
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(XML_OUTPUT_PATH + outputFileName + ".xml"));
+            writer.write(xml);
+            writer.close();
```

Comment:
Will this file be closed if an exception is thrown? Close always needs to be in a finally {} clause. You can use a try-with-resources to avoid needing to do it yourself
```suggestion
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(XML_OUTPUT_PATH + outputFileName + ".xml"))) {
            writer.write(xml);
```

is the same as:

```
try {
            BufferedWriter writer = new BufferedWriter(new FileWriter(XML_OUTPUT_PATH + outputFileName + ".xml"));
            writer.write(xml);
            ...
} finally {
          writer.close();
}